server/routes/comparison/abstractComparisonController.ts

```diff
@@ -0,0 +1,20 @@
+import { Request } from 'express'
+import HistoricalPrisonerService from '../../services/historicalPrisonerService'
+import { PrisonerDetailDto } from '../../@types/historical-prisoner/historicalPrisonerApiTypes'
+
+export default abstract class AbstractComparisonController {
+  protected constructor(protected readonly historicalPrisonerService: HistoricalPrisonerService) {}
+
+  protected async getPrisonerDetail(req: Request): Promise<PrisonerDetailDto[]> {
+    const { shortlist } = req.session
+    const results: PrisonerDetailDto[] = []
+
+    if (shortlist) {
```

Comment:
will probably need the check to ensure that either 2 or 3 items are in the shortlist